charts/linstor-scheduler-extender/templates/rbac.yaml

```
@@ -0,0 +1,71 @@
+{{- if .Values.rbac.create }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ include "linstor-scheduler-extender.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+rules:
+  - apiGroups:
+      - coordination.k8s.io
+    resources:
+      - leases
+    verbs:
+      - create
+      - get
+      - update
```

In my cluster kube-scheduler didn't started, trying to get configmap

```
I0318 19:45:02.322475       1 serving.go:347] Generated self-signed cert in-memory
W0318 19:45:02.496446       1 options.go:335] Neither --kubeconfig nor --master was specified. Using default API client. This might not work.
couldn't get policy config map linstor/linstor-scheduler-linstor-scheduler-extender: configmaps "linstor-scheduler-linstor-scheduler-extender" is forbidden: User "system:serviceaccount:linstor:linstor-scheduler-linstor-scheduler-extender" cannot get resource "configmaps" in API group "" in the namespace "linstor"
```

```suggestion
rules:
  - apiGroups: ["coordination.k8s.io"]
    resources: ["leases"]
    verbs: ["create", "get", "update"]
  {{- if semverCompare "< 1.22-0" (include "linstor-scheduler-extender.kubeVersion" .) }}
  - apiGroups: [""]
    resources: ["configmaps"]
    verbs: ["get", "list", "watch"]
  {{- end }}
```

---

Hmm, maybe we can just `--leader-elect-resource-lock=leases`? Looks like the default was changed in various releases.

---

No, because kube-scheduler before 1.22 requires to have permissions to read `linstor-scheduler-extender` configmap for itself.

This is policy configmap, not for the leaderelection one